suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -210,6 +210,12 @@ const getBundleThunk = createThunk(
                     configNetwork.accountType === 'legacy') &&
                 !discovery.availableCardanoDerivations?.includes(configNetwork.accountType);
 
+            // solana has a derivation path without account index, so we need to ensure that only single account gets discovered
+            const hasPathAccountIndex = configNetwork.bip43Path.includes('i');
+            if (!hasPathAccountIndex && !!prevAccounts[0]) {
+                return;
+            }
+
```

Comment:
In other words, if the path does not have `i` and there is already an account, stop discovering